packages/engine/src/datastore/arrow/meta_conversion.rs

```diff
@@ -182,7 +161,7 @@ impl HashDynamicMeta for RecordBatch<'_> {
 
 /// Computes the flat_buffers builder from the metadata, using this builder
 /// the metadata of a shared batch can be modified
-pub fn get_dynamic_meta_flatbuffers<'a>(meta: &DynamicMeta) -> Result<FlatBufferWrapper<'a>> {
+pub fn get_dynamic_meta_flatbuffers(meta: &DynamicMeta) -> Result<Vec<u8>> {
     // The reason why we're returning `FlatBufferBuilder` is that we don't want to
```

Comment:
The comment is reasoning correctly, that this may introduce an unneeded copy of the data. I'd track this in [this Asana task](https://app.asana.com/0/1201793759102797/1201849814189759/f) (_internal_)